test/puppetlabs/puppetdb/query/ast_test.clj

```diff
@@ -0,0 +1,132 @@
+(ns puppetlabs.puppetdb.query.ast-test
+  "Tests the behavior of AST queries."
+  (:require
+   [clj-time.coerce :as coerce-time]
+   [clj-time.core :refer [now]]
+   [clojure.test :refer :all]
+   [puppetlabs.kitchensink.core :refer [dissoc-in]]
+   [puppetlabs.puppetdb.cli.services :as cli-svcs]
+   [puppetlabs.puppetdb.command :refer [enqueue-command]]
+   [puppetlabs.puppetdb.examples :as examples]
+   [puppetlabs.puppetdb.testutils :refer [block-until-results]]
+   [puppetlabs.puppetdb.testutils.cli :refer [get-catalogs get-factsets]]
+   [puppetlabs.puppetdb.testutils.services :as svc-utils :refer [*server*]]
+   [puppetlabs.trapperkeeper.app :refer [get-service]]))
+
+(deftest ast-queries
```

Comment:
Happy to change it we prefer, but I did this initially because I didn't want an http test, i.e. that part of the stack's not relevant/desirable here -- these are intended to test AST forms as directly as possible, and because I was thinking about having additional tests eventually, which might or might not involve facts, and probably would cross resource types arbitrarily.